server/src/main/java/org/opensearch/indices/tiering/HotToWarmTieringService.java

```diff
@@ -0,0 +1,394 @@
+            indexTieringInfo.markSuccessful();
+            if (tieringRequestContext.isRequestProcessingComplete()) {
+                logger.info("[HotToWarmTiering] Tiering is completed for the request [{}]", tieringRequestContext);
+                completeRequestLevelTiering(tieringRequestContext);
+            }
+        }
+    }
+
+    IndexMetadata findIndexMetadata(final Metadata.Builder metadataBuilder, final Index index) {
+        IndexMetadata indexMetadata;
+        try {
+            indexMetadata = metadataBuilder.getSafe(index);
+        } catch (IndexNotFoundException e) {
+            return null;
+        }
+        return indexMetadata;
+    }
+
+    /**
+     * Updates the index metadata with the tiering settings/metadata for an accepted index.
+     * Accepted index is an index to be tiered from hot to warm.
+     * @param metadataBuilder metadata builder
+     * @param routingTableBuilder routing builder
+     * @param indexTieringInfo index tiering info
+     * @return updated index metadata builder
+     */
+    IndexMetadata.Builder updateIndexMetadataForAcceptedIndex(
+        final Metadata.Builder metadataBuilder,
+        final RoutingTable.Builder routingTableBuilder,
+        final TieringRequestContext.IndexTieringInfo indexTieringInfo
+    ) {
+        final Index index = indexTieringInfo.getIndex();
+        final IndexMetadata indexMetadata = findIndexMetadata(metadataBuilder, index);
+        if (indexMetadata == null) {
+            indexTieringInfo.markFailed("index not found");
+            return null;
+        } else if (!isHotIndex(indexMetadata)) {
+            indexTieringInfo.markFailed("index is not in the HOT tier");
+            return null;
+        }
+        Settings.Builder indexSettingsBuilder = Settings.builder().put(indexMetadata.getSettings());
+        // update index settings here
+        indexSettingsBuilder.put(INDEX_STORE_LOCALITY_SETTING.getKey(), IndexModule.DataLocalityType.PARTIAL);
+        indexSettingsBuilder.put(INDEX_TIERING_STATE.getKey(), IndexModule.TieringState.HOT_TO_WARM);
+
+        // Update number of replicas to 1 in case the number of replicas is greater than 1
+        if (Integer.parseInt(indexMetadata.getSettings().get(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey())) > 1) {
+            final String[] indices = new String[] { index.getName() };
+            routingTableBuilder.updateNumberOfReplicas(1, indices);
+            metadataBuilder.updateNumberOfReplicas(1, indices);
+        }
+        // trying to put transient index metadata in the custom index metadata
+        IndexMetadata.Builder indexMetadataBuilder = IndexMetadata.builder(indexMetadata).settings(indexSettingsBuilder);
+        final Map<String, String> tieringCustomData = new HashMap<>();
+        tieringCustomData.put(HOT_TO_WARM_START_TIME, String.valueOf(System.currentTimeMillis()));
+        tieringCustomData.put(HOT_TO_WARM_END_TIME, "-1");
```

Comment:
yes, i was thinking in terms of having information on the time taken for an index to be tiered, however thats more of a stats use-case. Also, for status api, since we display info about the in-progress/failed tierings, it does make sense to remove the end time from the custom-metadata.